File: CLASS2/10814.py
import sys
input = sys.stdin.readline

n = int(input())
name_arr = []

for idx in range(n):
    age, name = input().strip().split()
    name_arr.append((int(age), name, idx))

name_arr.sort(key = lambda x:(x[0], x[2]))
for age, name, _ in name_arr:
    print(age, name)

File: solved.ac/CLASS2_JS/10814.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const solution = () => {
  const n = Number(input());
  const arr = Array.from({ length: n }, () => input()).sort((a, b) => {
    const [age, name] = a.split(" ");
    const [age2, name2] = b.split(" ");
    return Number(age) - Number(age2);
  });
  console.log(arr.join("\n"));
};

solution();
